Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt

```diff
@@ -83,12 +84,25 @@ data class VaccinatedPerson(
     fun getDaysUntilImmunity(nowUTC: Instant = Instant.now()): Int? {
         val newestFullDose = getNewestFullDose() ?: return null
         val today = nowUTC.toLocalDateUserTz()
-        return IMMUNITY_WAITING_DAYS - Days.daysBetween(newestFullDose.vaccinatedOn, today).days
+        return if (isSeriesCompletingOverTwoWeeks(nowUTC)) 0
+        else IMMUNITY_WAITING_DAYS - Days.daysBetween(newestFullDose.vaccinatedOn, today).days
     }
 
     private fun getNewestDoseVaccinatedOn(): LocalDate =
         vaccinationCertificates.maxOf { it.vaccinatedOn }
 
+    private fun isSeriesCompletingOverTwoWeeks(nowUTC: Instant = Instant.now()): Boolean {
+        val certificate = vaccinationCertificates
+            .filter { it.isSeriesCompletingShot }
+            .firstOrNull {
+                Days.daysBetween(
+                    it.rawCertificate.vaccination.vaccinatedOn,
+                    nowUTC.toLocalDateUtc()
```

Comment:
Can you move that into the fun -> eg. (today: LocalDate) so it is recalculated for every certificate?

Reply:
isSeriesCompletingShotOverTwoWeeksOld(today: LocalDate) ?

Reply:
Yes